backend/src/junior/junior.controller.ts

```diff
@@ -0,0 +1,33 @@
+import { Controller, UsePipes, ValidationPipe, Post, Body, UseGuards, forwardRef, Inject } from '@nestjs/common';
+import { JuniorService } from './junior.service';
+import { LoginJuniorDto, RegisterJuniorDto } from './dto';
+import { AuthGuard } from '@nestjs/passport';
+import { Admin } from '../admin/admin.decorator';
+import { AuthenticationService } from '../authentication/authentication.service';
+import { AdminService } from '../admin/admin.service';
+
+@Controller('junior')
+export class JuniorController {
+
+    // Todo remove admin dependancy once guards are in place.
+    constructor(
+        private readonly juniorService: JuniorService,
+        private readonly authenticationService: AuthenticationService,
+        @Inject(forwardRef(() => AdminService))
```

Comment:
Are `forwardRef`s needed in controller?

edit: this will be removed later i guess anyways

Reply:
Correct you are, will change.